File: RA 2.1/Actividad 2.py
class Volador: 
    def moverse(self):
        print("volando")

class Nadador: 
    def moverse(self): 
        print("nadando")

#Clase sin sobrescritura: 
class Pato(Volador, Nadador): 
    pass

print("======Sin sobrescritura======")
p= Pato()
p.moverse()

print("Orden MRO: ", Pato.__mro__)

#Clase con sobrescritura y llamadas explicitas: 
class PatoExplicito(Volador, Nadador): 
    def moverse(self): 
        Volador.moverse(self)
        Nadador.moverse(self)
        print("Tambien camino")

print("======Con sobrescritura explicita======")
p2= PatoExplicito()
p2.moverse()

File: RA 2.1/Actividad 2.cpp
#include <iostream> 
using namespace std;

class Volador { 
    public: 
        virtual void moverse() {
            cout<<"volando"<<endl;
        }
};

class Nadador { 
    public: 
        virtual void movrese() {
            cout<<"nadando"<<endl;
        }
};

//Clase sin sobrescritura
class Pato : public Volador, public Nadador {

};

//Clase con sobrescritura
class PatoExplicito : public Volador, public Nadador { 
    public: 
        void moverse() override { 
            Volador::moverse(); 
            Nadador::movrese(); 
            cout<<"Tambien camino"<<endl;
        }
};

int main() {
    cout<<"======Sin sobrescritura======"<<endl;
    Pato p;
    p.moverse();

    cout<<"======Con sobrescritura explicita======"<<endl;
    PatoExplicito p2; 
    p2.moverse();
}

